dev-docs/RFCs/v7.1/animation-timeline-rfc.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Hiearchical Animation Timeline in luma.gl
+
+* **Author**: Tarek Sherif
+* **Date**: April, 2019
+* **Status**: **Draft**
+
+
+## Summary
+
+This RFC specifies a timeline management system to facilate more complex animations that can be easily controlled by the application.
+
+
+## Background
+
+Currently, the only support for animation provided by luma.gl is the passing of elapsed wall time and frame counts as `animationProps`. This pushes all orchestration of animation to the application, which can end up being quite complex. Since systems like deck.gl's transitions and luma.gl's uniform animations track `animationProps` time, there is no way to control them independantly of wall time.
+
+
+## Customers
+
+The Elevate team has requested the ability to control transitions in deck.gl, specifically, the ability to pause, play and scrub through a transition.
+
+
+## Overview
+
+A timeline manager that can provide `time` values to be used in animations that are independent of wall time. The timeline manager should support the following features:
+
+- play: provide a `time` value that elapses at the same rate as wall time
+- pause: `time` remains constant at the current value
+- set: set `time` to a specific value
+- multiple `channels` that provide `channelTime` values, related to `time`, but with the following additional properties:
+  * `rate`: a scaling factor that indicates how quickly `channelTime` elapses relative to `time`
+  * `startTime`: an offset into `time` at which `channelTime` begins elapsing
+  * `duration`: how long `channelTime` runs for
+  * `wrapMode`: what to do when `duration` elapses. Options are `clamp` (stop elapsing) or `loop` (go back to 0)
```

Comment:
I think it's more common to define the loop count? E.g. [animation-iteration-count](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-iteration-count) and [tween.js](https://github.com/tweenjs/tween.js/blob/master/docs/user_guide.md#repeat)

Reply:
Good call. Went with the tween.js naming for `delay` as well.